src/Symfony/Component/Console/Output/StreamOutput.php

```diff
@@ -81,31 +81,29 @@ protected function doWrite($message, $newline)
      *
      * Colorization is disabled if not supported by the stream:
      *
-     *  -  the stream is redirected (eg php file.php >log)
-     *  -  Windows without VT100 support, Ansicon, ConEmu, Mintty
-     *  -  non tty consoles
+     * This is tricky on Windows, because Cygwin, Msys2 etc emulate pseudo
+     * terminals via named pipes, so we can only check the environment.
      *
      * @return bool true if the stream supports colorization, false otherwise
      */
     protected function hasColorSupport()
     {
-        if (function_exists('stream_isatty') && !@stream_isatty($this->stream)) {
-            return false;
-        }
         if (DIRECTORY_SEPARATOR === '\\') {
-            if (function_exists('sapi_windows_vt100_support')) {
-                $vt100Enabled = @sapi_windows_vt100_support($this->stream);
-            } else {
-                $vt100Enabled = '10.0.10586' === PHP_WINDOWS_VERSION_MAJOR.'.'.PHP_WINDOWS_VERSION_MINOR.'.'.PHP_WINDOWS_VERSION_BUILD;
-            }
-
-            return
-                $vt100Enabled
+            return (function_exists('sapi_windows_vt100_support')
+                && @sapi_windows_vt100_support($this->stream))
                 || false !== getenv('ANSICON')
                 || 'ON' === getenv('ConEmuANSI')
                 || 'xterm' === getenv('TERM');
         }
 
-        return function_exists('posix_isatty') && @posix_isatty($this->stream);
+        if (function_exists('stream_isatty')) {
+            return @stream_isatty($this->stream);
+        } elseif (function_exists('posix_isatty')) {
```

Comment:
just `if`, not `elseif` as the previous case returns

Reply:
Okay